riscv/eviction_policies.cc

```diff
@@ -0,0 +1,90 @@
+#include "eviction_policies.h"
+#include <stdio.h>
+
+uint32_t lfsr_t::next(size_t idx)
+{
+    reg = (reg>>1)^(-(reg&1) & 0xd0000001);
+    return reg%ways;
+}
+
+void lfsr_t::update(uint64_t addr, uint32_t way, const size_t idx_shift) {}
+
+void lfsr_t::insert(size_t idx, size_t way) {}
+
+uint32_t lru_t::lowest_recency_way(size_t idx)
+{
+    size_t argmin = 0;
+    uint32_t min_value = 0xffffffff; // Abitrarily big
+    for (size_t i = 0; i < ways; i++) {
+      if (min_value > this->metadata[idx*ways+i]) {
+        min_value = this->metadata[idx*ways+i];
+        argmin = i;
+      }
+    }
+    return argmin;
```

Comment:
I suggest using `min_element` to compute argmin.  It returns an iterator positioned at the minimum element, so then you can use iterator arithmetic to retrieve the index.

    auto begin = this->metadata + idx * ways;
    return std::min_element(begin, begin + ways) - begin;